src/Neo/SmartContract/Native/ContractEventAttribute.cs

```diff
@@ -17,42 +17,62 @@ namespace Neo.SmartContract.Native
 {
     [DebuggerDisplay("{Descriptor.Name}")]
     [AttributeUsage(AttributeTargets.Constructor, AllowMultiple = true)]
-    internal class ContractEventAttribute : Attribute
+    internal class ContractEventAttribute : Attribute, IHardforkActivable
     {
         public int Order { get; init; }
         public ContractEventDescriptor Descriptor { get; set; }
         public Hardfork? ActiveIn { get; init; } = null;
+        public Hardfork? DeprecatedIn { get; init; } = null;
 
         public ContractEventAttribute(Hardfork activeIn, int order, string name,
             string arg1Name, ContractParameterType arg1Value) : this(order, name, arg1Name, arg1Value)
         {
             ActiveIn = activeIn;
         }
 
+        public ContractEventAttribute(Hardfork activeIn, int order, string name,
+            string arg1Name, ContractParameterType arg1Value, Hardfork deprecatedIn) : this(activeIn, order, name, arg1Name, arg1Value)
+        {
+            DeprecatedIn = deprecatedIn;
+        }
+
```

Comment:
Do you really need all these `constructors`? Unless their required and absolutely needed than yes, otherwise just make properties your setting `Public`. `Public` for optional.

Reply:
Yes because enum doesn't work as arguments :(